shared/src/main/scala/Checker.scala

```diff
@@ -0,0 +1,3 @@
+package carte
+
+abstract class Checker
```

Comment:
Which version of the main branch did you start from? Because IIUC the purpose of this `Checker` class is to be a common interface for `WhitelistChecker` and `BlacklistChecker`. But I already defined a trait with this purpose in my last PR. Or maybe it is something else?

Reply:
It's a common interface for semantic checkers and syntactic checkers.

Reply:
The module is no longer related to this pull request. I will remove it. It was introduced because of the previous implementation of `testkit.Specs`.